mem/qry.go

```diff
@@ -106,8 +120,23 @@ func (qry *memlogQuery) Next(ctx context.Context) (interface{}, error) {
 	qry.log.l.Lock()
 	defer qry.log.l.Unlock()
 
+	if qry.reverse {
+		if qry.live {
```

Comment:
Maybe this check should be done when applying the option to the query? Both in Live and Reverse.

Reply:
I tried that but only got it working for the normal, not Push mode.